src/components/color-picker/color-field/color-field.tsx

```diff
@@ -0,0 +1,494 @@
+      Math.min(canvasWidth, event.clientX - rect.left)
+    );
+    this.#currentY = Math.max(
+      0,
+      Math.min(canvasHeight, event.clientY - rect.top)
+    );
+
+    this.#colorPickedVariants = this.#getColorFromCanvas();
+    this.#drawMarker();
+  };
+
+  componentWillLoad() {
+    // Initialize colorVariants before component loads to avoid state change on load
+    if (this.value) {
+      try {
+        this.colorVariants = fromStringToColorVariants(this.value);
+        const format = getColorFormat(this.value);
+
+        if (format) {
+          this.#colorFormat = format;
+        }
+      } catch (error) {
+        console.warn(`Invalid value color: ${this.value}`, error);
+        this.colorVariants = fromStringToColorVariants("#000000");
+        this.#colorFormat = "hex";
+      }
+    }
+  }
+
+  connectedCallback() {
+    // Accessibility
+    this.internals.setFormValue(this.value);
+    const labels = this.internals.labels;
+    const accessibleNameFromExternalLabel = getElementInternalsLabel(labels);
+
+    // Report any accessibility issue
+    analyzeLabelExistence(
+      this.el,
+      "ch-color-field",
+      labels,
+      accessibleNameFromExternalLabel,
+      this.translations.accessibleName.label
+    );
+  }
+
+  componentDidLoad() {
+    this.#updateCanvasSize();
+    this.#updateMarkerPosition();
+
+    // Initialize ResizeObserver with RAF optimization
+    this.#resizeObserver = new ResizeObserver(() => {
+      this.#resizeRAF.perform(() => {
+        this.#updateCanvasSize();
+        this.#updateMarkerPosition();
+      });
+    });
+
+    // Observe the canvas for size changes
+    this.#resizeObserver.observe(this.#canvasRef);
+  }
+
+  disconnectedCallback() {
+    // Disconnect the observer when the component is unloaded
+    if (this.#resizeObserver) {
+      this.#resizeObserver.disconnect();
+    }
+
+    // Cancel pending RAF operations
+    this.#resizeRAF.cancel();
+    this.#dragRAF?.cancel();
+
+    // Clean up event listeners
+    document.removeEventListener("mousemove", this.#handleMouseMove, {
+      capture: true
+    });
+    document.removeEventListener("mouseup", this.#handleMouseUp, {
+      capture: true
+    });
+  }
+
+  render() {
+    const { accessibleName } = this.translations;
+    const isInteractive = !this.disabled && !this.readonly;
+
+    return (
+      <Host
+        aria-label={accessibleName.label}
+        aria-roledescription={accessibleName.description}
+      >
+        <canvas
+          aria-disabled={this.disabled ? "true" : null}
+          aria-readonly={this.readonly ? "true" : null}
+          onClick={isInteractive ? this.#handleCanvasClick : null}
+          onMouseDown={isInteractive ? this.#handleMouseDown : null}
+          onKeyDown={isInteractive ? this.#handleKeyDown : null}
+          ref={el => (this.#canvasRef = el as HTMLCanvasElement)}
+          role="application"
+          tabindex={isInteractive ? "0" : "-1"}
+        ></canvas>
```

Comment:
The semantic states (aria attributes) must be placed in the same element that has the role.

```suggestion
        role="application"
        aria-disabled={this.disabled ? "true" : null}
        aria-label={this.accessibleName}
        aria-readonly={this.readonly ? "true" : null}
        aria-roledescription={this.accessibleRoleDescription}
        tabindex={!this.disabled ? "0" : null}
      >
        <canvas
          aria-hidden="true"
          onClick={isInteractive ? this.#handleCanvasClick : null}
          onMouseDown={isInteractive ? this.#handleMouseDown : null}
          onKeyDown={isInteractive ? this.#handleKeyDown : null}
          ref={el => (this.#canvasRef = el as HTMLCanvasElement)}
        ></canvas>
```